src/main/java/nl/tudelft/cse1110/andy/result/GenericFailure.java

```diff
@@ -0,0 +1,59 @@
+package nl.tudelft.cse1110.andy.result;
+
+public class GenericFailure {
+    private final String genericFailureMessage;
+    private final String stepName;
+    private final String exceptionMessage;
+    private final Integer externalProcessExitCode;
+    private final String externalProcessErrorMessages;
+
+    private GenericFailure(String genericFailureMessage, String stepName, String exceptionMessage, Integer externalProcessExitCode, String externalProcessErrorMessages) {
+        this.genericFailureMessage = genericFailureMessage;
+        this.stepName = stepName;
+        this.exceptionMessage = exceptionMessage;
+        this.externalProcessExitCode = externalProcessExitCode;
+        this.externalProcessErrorMessages = externalProcessErrorMessages;
+    }
+
+    public static GenericFailure build(String genericFailure, String stepName, String exceptionMessage, Integer externalProcessExitCode, String externalProcessErrorMessages) {
+        boolean genericFailureHappened = genericFailure != null || stepName != null || exceptionMessage != null;
+        boolean externalProcessFailed = externalProcessExitCode != null || externalProcessErrorMessages != null;
+
+        if (!genericFailureHappened && !externalProcessFailed) {
+            return null;
```

Comment:
Should we return null or should we return a "NoFailure" object to avoid returning null?

Reply:
Or return a Optional ? At least it becomes explicit that this can be null.

Reply:
I made it so that a GenericFailure object is always created (it is never null in a valid result), and added a `hasFailure` method in GenericFailure, which checks if any of its internal fields are non-null.